### PR TITLE
Implement rudimentary beatmap designer

### PR DESCRIPTION
**Pull request by petermonky**

### CODE REVIEW

File: PulsePace/BeatmapDesigner/Views/BeatmapDesignerView.swift [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                viewModel.audioManager.togglePlayer()
+            }
+
+            PlaybackControlButtonView(systemName: "goforward.10") {
+
+            }
+        }
+    }
+
+    func renderHitObject(_ hitObject: any HitObject) -> some View {
```
Comment (ckcherry23) [inline PulsePace/BeatmapDesigner/Views/BeatmapDesignerView.swift:88]: We can refactor these render functions to separate files later and can also reuse some for the gameplay screen, am I right?

File: PulsePace/BeatmapDesigner/Views/BeatmapDesignerView.swift [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    func renderSlider(player: AVAudioPlayer) -> some View {
+        ZStack {
+            Rectangle()
+                .foregroundColor(.blue)
+                .frame(width: 2 * player.duration, height: 20)
+                .offset(x: player.duration - 2 * viewModel.sliderValue)
+                .gesture(
+                    DragGesture()
```
Comment (ckcherry23) [inline PulsePace/BeatmapDesigner/Views/BeatmapDesignerView.swift:52]: Will see what I can do to about abstracting these gestures

File: PulsePace/BeatmapDesigner/Views/PlaybackControlButtonView.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+//  Created by Peter Jung on 2023/03/11.
+//
+
+import SwiftUI
+
+struct PlaybackControlButtonView: View {
+    var systemName: String = "play"
+    var fontSize: CGFloat = 24
+    var color: Color = .accentColor
+    var action: () -> Void
```
Comment (ckcherry23) [inline PulsePace/BeatmapDesigner/Views/PlaybackControlButtonView.swift:14]: We can consider reusing `Command` here

File: PulsePace/BeatmapDesigner/Views/BeatmapDesignerView.swift (hunk 1 above)
Comment (petermonky) [reply PulsePace/BeatmapDesigner/Views/BeatmapDesignerView.swift:88]: correct we can create separate view files for them to reuse with the gameplay 👍

**Pull request merged**